src/scripts/types/mhct.ts

```diff
@@ -0,0 +1,36 @@
+export interface IntakeMessage {
+    extension_version: number;
+    user_id: number;
+    entry_id: number;
+    location: ComponentEntry | null;
```

Comment:
I really want this to not be nullable so the rules can depend on location being set. I did make it nullable _for now_ because a lot of the stage funcs will set the location to null.

Previously, the `recordHuntWithPrehuntUser` would check that location is null after the stage function so it could fail out. I think it may be better for the stage funcs to throw upon encountering an error (since it now wrapped in a try/catch) as we can assure location is _not_ null.

This null check on location is temporarily in `engine.validateMessage`.

Reply:
in a separate pr we can fix stage checks to not use location. thanks for explaining.